### PR TITLE
Level entity

### PR DESCRIPTION
**Pull request by Teagan42**
Implement Enumerated State Device UI Elements:

- Retrieve enumerated state options from the state object, option properties to display label/icon and value
- Reactive UI elements to device state changes
- Fan domain informational state card

@james-gibson

### CODE REVIEW

**Comment by Teagan42** [comment]
@james-gibson Please review more-info-levels.js to see if you can reduce the complexity of it.

File: src/more-infos/more-info-levels.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<link rel='import' href='../../bower_components/polymer/polymer.html'>
+<link rel='import' href='../../bower_components/paper-button/paper-button.html'>
+<link rel='import' href='../../bower_components/iron-icon/iron-icon.html'>
+
+<dom-module id='more-info-levels'>
+  <style is="custom-style" include="iron-flex"></style>
+  <style>
```
Comment (james-gibson) [inline src/more-infos/more-info-levels.html]: extraneous?

File: src/util/state-more-info-type.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const DOMAINS_WITH_MORE_INFO = [
   'light', 'group', 'sun', 'configurator', 'thermostat', 'script',
```
Comment (james-gibson) [inline src/util/state-more-info-type.js]: newlines

File: src/more-infos/more-info-levels.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      type: Array,
+    },
+  },
+
+  stateObjChanged(stateObj) {
+    this.selectedOption = this.computeSelected(stateObj);
+    this.options = stateObj.attributes.options;
+  },
+
+  findSelected(optionVal) {
```
Comment (james-gibson) [inline src/more-infos/more-info-levels.js]: `return this.stateObj.attributes.options.filter(x => {return x.value == optionVal ? x : null}) || {}`

Comment (james-gibson) [inline src/more-infos/more-info-levels.js]: i think

Comment (james-gibson) [inline src/more-infos/more-info-levels.js]: Argh you will need to get the value from the array...my bad

Comment (Teagan42) [inline src/more-infos/more-info-levels.js]: yeah can't use filter for it really

**Pull request merged**